### PR TITLE
Add missing npmignores and fix a few others.  Alphabetize and cleanup deps

### PR DESCRIPTION
**Pull request by calebmshafer**
Ended up reviewing the files published in our packages when switching around builds and noticed a few of them had issues.  This is a rather large PR so listing out the main changes;

- The list of npmignore changes is [below](#updates-to-files-included-in-published-packages).
- The other main change in this PR is alphabetizing all of the dependencies and some other clean up in `package.json` files.
  - Rush does most of this automatically when you publish so while testing it cleaned it up
  - I fixed a few `peerDeps` that were missing a `^` in as their semver string.

## Updates to files included in publish packages

- @bentley/itwin-client
  - Remove incorrectly add coverage files and other test files
- @bentley/imodelhub-client-tests
  - Publishing test iModel from both `src/assets` and `lib/assets`. Switch to only publish the `lib/assets` files. 
  - @Animagne can you review the files excluded here to make sure it still has everything it needs to run when published?  If you want, feel free to reach out for a full diff of what ends up in the package.  I have it locally but it's rather large. 
- @bentley/geonames-extension
  - Missing a npmignore and publishing build artifacts, including .rush files 
- @bentley/extension-webpack-tools
  - Missing a npmignore and publishing build artifacts, including .rush files 
- @bentley/express-server
  - Missing a npmignore and publishing test files
- @bentley/eslint-plugin
  - Missing a npmignore

### CODE REVIEW

**Review by grigasp — APPROVED** [review]

**Review by Animagne — APPROVED** [review]

**Review by pauliunas — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Review by scsewall — APPROVED** [review]

**Review by EarlinLutz — APPROVED** [review]
I don't know why both flatbuffer 1.10 and 1.12 are present.   (in core/geometry).

The native 1.12 schema compiler was used to create BGFBAccessors.ts, so 1.12 would seem to be prefered.

You can resolve as you feel best.

**Comment by calebmshafer** [comment]
> I don't know why both flatbuffer 1.10 and 1.12 are present. (in core/geometry).
> 
> The native 1.12 schema compiler was used to create BGFBAccessors.ts, so 1.12 would seem to be prefered.
> 
> You can resolve as you feel best.

Looks like it is only the @types package that is still on 1.10 so it's not too big of deal. I'll open a separate PR to fix it though.

**Comment by EarlinLutz** [comment]
Good.

**Review by ColinKerr — APPROVED** [review]

**Comment by calebmshafer** [comment]
@pmconne mind taking a look?

**Review by pmconne — APPROVED** [review]

File: tools/extension-webpack/.npmignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# start off ignoring everything, and then add back only the files we want
+*
+!bin/**/*.js
+!config/**/*.js
+!scripts/**/*.js
+!*.md
+!index.d.ts
```
Comment (wgoehrig) [inline tools/extension-webpack/.npmignore:7]: nit: should end in newline

**Review by wgoehrig — APPROVED** [review]

File: tools/extension-webpack/.npmignore (hunk 1 above)
Comment (wgoehrig) [reply tools/extension-webpack/.npmignore:7]:
```suggestion
!index.d.ts

```

Comment (calebmshafer) [reply tools/extension-webpack/.npmignore]: fixed, thanks

**Review by ramanujam-raman — APPROVED** [review]

**Review by swwilson-bsi — APPROVED** [review]

**Pull request merged**